### ISSUE TITLE
Question about the COCO2017 pre-training

### ISSUE THREAD

**Issue by zhengziqiang** [RobinWu218/SimDINO#17]
Dear authors, thank you for releasing your awesome work!

I have noticed Figure 4 of your work, performing pre-training on the COCO2017 training images (118,287 images in total), and conducting the KNN evaluation on the ImageNet-1K dataset. 

I used the official DINO codes to run the experiments on the COCO2017 training images. I used the official hyper-parameters as follows:

total batch size: 224,
basic_lr: 0.0005,
n_local_crops: 8,
patch_size: 16,
out_dim: 65536,
norm_last_layer: True,
momentum_teacher: 0.996,
warmup_teacher_temp: 0.04,
teacher_temp: 0.04,
use_fp16: True,
weight_decay: 0.04,
weight_decay_end: 0.4,
global_crops_scale: (0.4, 1.),
local_crops_scale: (0.05, 0.4),
drop_path_rate: 0.1,
freeze_last_layer: 1,
arch: vit_base,
warmup_teacher_temp_epochs: 0,
clip_grad: 3.0

I could achieve K-NN accuracy (using the official [eval_knn.py](https://github.com/facebookresearch/dino/blob/main/eval_knn.py) of DINO) on the ImageNet-1K dataset as follows at even the epoch of **60**:

10-NN classifier result: Top1: 34.536, Top5: 52.382
20-NN classifier result: Top1: **35.272**, Top5: **55.416**
100-NN classifier result: Top1: 34.182, Top5: 58.488
200-NN classifier result: Top1: 32.796, Top5: 58.192

This is much higher than the results you reported (even higher than your SimDINO method at the epoch 100) in Figure 4.

Could you please provide more implementation details about these very weird results? 

The results should not be so slow.

**Comment by zhengziqiang** [comment]
<img width="458" alt="Image" src="https://github.com/user-attachments/assets/ac3ccd8d-675b-4348-94f7-244ce1454db6" />

**Comment by RobinWu218** [comment]
Hi @zhengziqiang thanks for your interest in our work and this interesting observation. Our setting on DINOv1 mostly follows the official setting on ViT-B/16 ([source](https://dl.fbaipublicfiles.com/dino/dino_vitbase16_pretrain/args.txt)):

lr: 0.00075,
warmup_epochs: 10, 
min_lr: 2e-06
n_local_crops: 10,
patch_size: 16,
out_dim: 65536,
norm_last_layer: True,
momentum_teacher: 0.996,
warmup_teacher_temp: 0.04,
teacher_temp: 0.07,
use_fp16: True,
weight_decay: 0.04,
weight_decay_end: 0.4,
global_crops_scale: (0.4, 1.),
local_crops_scale: (0.05, 0.4),
drop_path_rate: 0.1,
freeze_last_layer: 3,
arch: vit_base,
warmup_teacher_temp_epochs: 10,
clip_grad: 3.0,
batch_size_per_gpu: 128,
world_size: 8

We use this setting as it performs better on the ImageNet pretraining experiments than the default values. Compared to your setting, the main differences seem to come from the data processing (e.g. `n_local_crops`) and optimization (e.g. lr and teacher temp schedules). Also, your total batch size is 224, which is a bit odd. Do you mean the resolution size?

**Comment by zhengziqiang** [comment]
No, It is the batch size. I run the experiments on a single H800 GPU. The batch size is 224*1. I feel the key difference is teacher temp, which you are using 0.07.

**Comment by zhengziqiang** [comment]
Here are my used parameters and the corresponding training log:

arch: vit_base
batch_size_per_gpu: 224
clip_grad: 3.0
data_path: ./data/coco/train/
dist_url: env://
drop_path_rate: 0.1
epochs: 100
freeze_last_layer: 1
global_crops_scale: (0.4, 1.0)
gpu: 0
local_crops_number: 8
local_crops_scale: (0.05, 0.4)
lr: 0.0005
min_lr: 1e-06
momentum_teacher: 0.996
norm_last_layer: True
num_workers: 10
optimizer: adamw
out_dim: 65536
output_dir: ./output/coco_vitb
patch_size: 16
rank: 0
saveckp_freq: 20
seed: 0
teacher_temp: 0.04
use_bn_in_head: False
use_fp16: True
warmup_epochs: 10
warmup_teacher_temp: 0.04
warmup_teacher_temp_epochs: 0
weight_decay: 0.04
weight_decay_end: 0.4
world_size: 1
Data loaded: there are 118287 images.


{"train_loss": 10.847061585296284, "train_lr": 2.1837706004925177e-05, "train_wd": 0.04002952329616741, "epoch": 0}
{"train_loss": 10.216308844811989, "train_lr": 6.559599355938625e-05, "train_wd": 0.040206964210518024, "epoch": 1}
{"train_loss": 10.1687733205882, "train_lr": 0.00010935428111384733, "train_wd": 0.040561839144516526, "epoch": 2}
{"train_loss": 9.684183140595755, "train_lr": 0.0001531125686683084, "train_wd": 0.0410937978794477, "epoch": 3}
{"train_loss": 9.242280662059784, "train_lr": 0.0001968708562227695, "train_wd": 0.041802315436264385, "epoch": 4}
{"train_loss": 8.810609721776212, "train_lr": 0.0002406291437772306, "train_wd": 0.04268669259367816, "epoch": 5}
{"train_loss": 8.347592439615365, "train_lr": 0.0002843874313316916, "train_wd": 0.04374605657820704, "epoch": 6}
{"train_loss": 7.906448284784953, "train_lr": 0.0003281457188861527, "train_wd": 0.04497936192549681, "epoch": 7}
{"train_loss": 7.47897985216343, "train_lr": 0.0003719040064406138, "train_wd": 0.04638539151206833, "epoch": 8}
{"train_loss": 7.12551839243282, "train_lr": 0.0004156622939950748, "train_wd": 0.04796275775647166, "epoch": 9}
{"train_loss": 6.789779375899922, "train_lr": 0.00043745580668762605, "train_wd": 0.04970990398866153, "epoch": 10}
{"train_loss": 6.505865402293928, "train_lr": 0.0004371902083272415, "train_wd": 0.0516251059862421, "epoch": 11}
{"train_loss": 6.261172365961653, "train_lr": 0.00043665908339412477, "train_wd": 0.053706473676066895, "epoch": 12}
{"train_loss": 6.053862383871367, "train_lr": 0.00043586307898219544, "train_wd": 0.05595195299951277, "epoch": 13}
{"train_loss": 5.8702974111744854, "train_lr": 0.0004348031649002129, "train_wd": 0.05835932793958727, "epoch": 14}
{"train_loss": 5.707239293690884, "train_lr": 0.0004334806324902185, "train_wd": 0.06092622270787065, "epoch": 15}
{"train_loss": 5.570143659006465, "train_lr": 0.00043189709305423477, "train_wd": 0.06365010408913147, "epoch": 16}
{"train_loss": 5.442593865322344, "train_lr": 0.0004300544758911375, "train_wd": 0.06652828394130456, "epoch": 17}
{"train_loss": 5.328384447278398, "train_lr": 0.00042795502594610734, "train_wd": 0.06955792184836285, "epoch": 18}
{"train_loss": 5.224833573355819, "train_lr": 0.0004256013010755062, "train_wd": 0.07273602792346456, "epoch": 19}
{"train_loss": 5.124500013662107, "train_lr": 0.00042299616893052677, "train_wd": 0.07605946575961099, "epoch": 20}
{"train_loss": 5.036160378745108, "train_lr": 0.000420142803463396, "train_wd": 0.07952495552490138, "epoch": 21}
{"train_loss": 4.944538546331001, "train_lr": 0.0004170446810604095, "train_wd": 0.08312907719933113, "epoch": 22}
{"train_loss": 4.863863609956972, "train_lr": 0.00041370557630648544, "train_wd": 0.08686827394993821, "epoch": 23}
{"train_loss": 4.7859836148493216, "train_lr": 0.00041012955738641775, "train_wd": 0.09073885564096884, "epoch": 24}
{"train_loss": 4.714290791388714, "train_lr": 0.0004063209811284169, "train_wd": 0.09473700247559454, "epoch": 25}
{"train_loss": 4.648138161861535, "train_lr": 0.00040228448769598585, "train_wd": 0.09885876876559196, "epoch": 26}
{"train_loss": 4.575710408615343, "train_lr": 0.0003980249949345992, "train_wd": 0.10310008682525959, "epoch": 27}
{"train_loss": 4.51362513683059, "train_lr": 0.0003935476923800613, "train_wd": 0.1074567709857332, "epoch": 28}
{"train_loss": 4.45283603758523, "train_lr": 0.0003888580349358597, "train_wd": 0.11192452172573529, "epoch": 29}
{"train_loss": 4.396982989527962, "train_lr": 0.000383961736227204, "train_wd": 0.11649892991468325, "epoch": 30}
{"train_loss": 4.338454807346517, "train_lr": 0.0003788647616398525, "train_wd": 0.12117548116396983, "epoch": 31}
{"train_loss": 4.280898026896246, "train_lr": 0.00037357332105221333, "train_wd": 0.1259495602821189, "epoch": 32}
{"train_loss": 4.227944378599976, "train_lr": 0.0003680938612695568, "train_wd": 0.13081645582942233, "epoch": 33}
{"train_loss": 4.177271829410032, "train_lr": 0.0003624330581695827, "train_wd": 0.13577136476756244, "epoch": 34}
{"train_loss": 4.127405990705346, "train_lr": 0.0003565978085688876, "train_wd": 0.14080939719963048, "epoch": 35}
{"train_loss": 4.080402622620265, "train_lr": 0.00035059522182025963, "train_wd": 0.14592558119586505, "epoch": 36}
{"train_loss": 4.034150856010841, "train_lr": 0.00034443261115102463, "train_wd": 0.1511148677003462, "epoch": 37}
{"train_loss": 3.9869473657824774, "train_lr": 0.00033811748475300464, "train_wd": 0.1563721355138047, "epoch": 38}
{"train_loss": 3.9434498232422452, "train_lr": 0.0003316575366349447, "train_wd": 0.16169219634762566, "epoch": 39}
{"train_loss": 3.9034676339590186, "train_lr": 0.00032506063724855005, "train_wd": 0.16706979994406518, "epoch": 40}
{"train_loss": 3.8597827933051367, "train_lr": 0.00031833482389954703, "train_wd": 0.17249963925761952, "epoch": 41}
{"train_loss": 3.8169482440659492, "train_lr": 0.00031148829095546904, "train_wd": 0.1779763556924381, "epoch": 42}
{"train_loss": 3.776030607295759, "train_lr": 0.00030452937986208173, "train_wd": 0.1834945443906128, "epoch": 43}
{"train_loss": 3.73663199580077, "train_lr": 0.00029746656898061325, "train_wd": 0.18904875956611636, "epoch": 44}
{"train_loss": 3.697052724885218, "train_lr": 0.00029030846325818146, "train_wd": 0.19463351987913977, "epoch": 45}
{"train_loss": 3.6606995752363494, "train_lr": 0.0002830637837439867, "train_wd": 0.20024331384551042, "epoch": 46}
{"train_loss": 3.6209740476174788, "train_lr": 0.00027574135696405903, "train_wd": 0.20587260527586143, "epoch": 47}
{"train_loss": 3.5850668946901956, "train_lr": 0.0002683501041674979, "train_wd": 0.21151583873918492, "epoch": 48}
{"train_loss": 3.5502632463520225, "train_lr": 0.0002608990304572994, "train_wd": 0.21716744504536678, "epoch": 49}
{"train_loss": 3.5150142849394768, "train_lr": 0.0002533972138190306, "train_wd": 0.22282184674131092, "epoch": 50}
{"train_loss": 3.4737960876840535, "train_lr": 0.000245853794060698, "train_wd": 0.22847346361520718, "epoch": 51}
{"train_loss": 3.444075457525976, "train_lr": 0.00023827796167730258, "train_wd": 0.23411671820352856, "epoch": 52}
{"train_loss": 3.405566937092579, "train_lr": 0.00023067894665363648, "train_wd": 0.23974604129531493, "epoch": 53}
{"train_loss": 3.373046693928314, "train_lr": 0.00022306600721896708, "train_wd": 0.2453558774283115, "epoch": 54}
{"train_loss": 3.3388596750570065, "train_lr": 0.00021544841856731213, "train_wd": 0.25094069037153843, "epoch": 55}
{"train_loss": 3.30648064252102, "train_lr": 0.000207835461557044, "train_wd": 0.25649496858888343, "epoch": 56}
{"train_loss": 3.2730887148416405, "train_lr": 0.00020023641140359434, "train_wd": 0.2620132306783228, "epoch": 57}
{"train_loss": 3.237574385874199, "train_lr": 0.00019266052637903003, "train_wd": 0.2674900307814005, "epoch": 58}
{"train_loss": 3.203562522024819, "train_lr": 0.00018511703653227877, "train_wd": 0.27291996395763835, "epoch": 59}
{"train_loss": 3.172054476358674, "train_lr": 0.0001776151324437328, "train_wd": 0.27829767151855495, "epoch": 60}
{"train_loss": 3.135563156369961, "train_lr": 0.00017016395402794478, "train_wd": 0.28361784631604553, "epoch": 61}
{"train_loss": 3.105818684353973, "train_lr": 0.00016277257939804752, "train_wd": 0.2888752379798979, "epoch": 62}
{"train_loss": 3.074217025980805, "train_lr": 0.0001554500138054745, "train_wd": 0.2940646580992723, "epoch": 63}
{"train_loss": 3.0454741282896562, "train_lr": 0.00014820517866844443, "train_wd": 0.29918098534303805, "epoch": 64}
{"train_loss": 3.008270114660263, "train_lr": 0.0001410469007025913, "train_wd": 0.30421917051390523, "epoch": 65}
{"train_loss": 2.9781361150019094, "train_lr": 0.0001339839011669683, "train_wd": 0.3091742415313792, "epoch": 66}
{"train_loss": 2.9465752695546006, "train_lr": 0.00012702478523853738, "train_wd": 0.3140413083385989, "epoch": 67}
{"train_loss": 2.9149655040466422, "train_lr": 0.00012017803152808834, "train_wd": 0.31881556772823216, "epoch": 68}
{"train_loss": 2.8839461880199835, "train_lr": 0.00011345198175035521, "train_wd": 0.323492308082667, "epoch": 69}
{"train_loss": 2.854929181662473, "train_lr": 0.00010685483056092397, "train_wd": 0.32806691402380095, "epoch": 70}
{"train_loss": 2.821126000899257, "train_lr": 0.0001003946155723056, "train_wd": 0.33253487096787027, "epoch": 71}
{"train_loss": 2.794977413433971, "train_lr": 9.407920756134404e-05, "train_wd": 0.33689176958079436, "epoch": 72}
{"train_loss": 2.766790489355723, "train_lr": 8.791630087988633e-05, "train_wd": 0.34113331012965653, "epoch": 73}
{"train_loss": 2.739238400802468, "train_lr": 8.191340408040096e-05, "train_wd": 0.3452553067260223, "epoch": 74}
{"train_loss": 2.708850721066648, "train_lr": 7.607783076796291e-05, "train_wd": 0.34925369145690566, "epoch": 75}
{"train_loss": 2.6831486374139786, "train_lr": 7.041669068975196e-05, "train_wd": 0.35312451839930864, "epoch": 76}
{"train_loss": 2.6564059176228265, "train_lr": 6.49368810729219e-05, "train_wd": 0.3568639675143695, "epoch": 77}
{"train_loss": 2.63227452518362, "train_lr": 5.964507822139054e-05, "train_wd": 0.36046834841728, "epoch": 78}
{"train_loss": 2.6044453418616094, "train_lr": 5.4547729381793025e-05, "train_wd": 0.3639341040192528, "epoch": 79}
{"train_loss": 2.5897707546299156, "train_lr": 4.965104488850388e-05, "train_wd": 0.367257814037933, "epoch": 80}
{"train_loss": 2.5608103505589743, "train_lr": 4.4960990597301774e-05, "train_wd": 0.3704361983728094, "epoch": 81}
{"train_loss": 2.540383286096833, "train_lr": 4.0483280616893036e-05, "train_wd": 0.37346612034227405, "epoch": 82}
{"train_loss": 2.520128509311965, "train_lr": 3.6223370347150525e-05, "train_wd": 0.37634458977915053, "epoch": 83}
{"train_loss": 2.4985340133760916, "train_lr": 3.218644983254896e-05, "train_wd": 0.3790687659816252, "epoch": 84}
{"train_loss": 2.486332486073176, "train_lr": 2.8377437438895456e-05, "train_wd": 0.38163596051668147, "epoch": 85}
{"train_loss": 2.4725702572049517, "train_lr": 2.4800973861058344e-05, "train_wd": 0.3840436398732535, "epoch": 86}
{"train_loss": 2.4551233477664716, "train_lr": 2.1461416468995215e-05, "train_wd": 0.38628942796250004, "epoch": 87}
{"train_loss": 2.4439245186971896, "train_lr": 1.8362833998968893e-05, "train_wd": 0.38837110846271283, "epoch": 88}
{"train_loss": 2.4279428088303767, "train_lr": 1.5509001596419113e-05, "train_wd": 0.3902866270065628, "epoch": 89}
{"train_loss": 2.414990549737757, "train_lr": 1.2903396216529194e-05, "train_wd": 0.3920340932085121, "epoch": 90}
{"train_loss": 2.4080402092500166, "train_lr": 1.0549192388091745e-05, "train_wd": 0.39361178253039153, "epoch": 91}
{"train_loss": 2.398035952990705, "train_lr": 8.449258345834444e-06, "train_wd": 0.39501813798331753, "epoch": 92}
{"train_loss": 2.3937504038666235, "train_lr": 6.606152535917747e-06, "train_wd": 0.3962517716642509, "epoch": 93}
{"train_loss": 2.388825906045509, "train_lr": 5.022120498862356e-06, "train_wd": 0.3973114661256851, "epoch": 94}
{"train_loss": 2.3811173475149907, "train_lr": 3.6990921337039712e-06, "train_wd": 0.39819617557712683, "epoch": 95}
{"train_loss": 2.377727810632099, "train_lr": 2.638679346708652e-06, "train_wd": 0.3989050269171585, "epoch": 96}
{"train_loss": 2.3768767470663246, "train_lr": 1.8421740875132916e-06, "train_wd": 0.39943732059508846, "epoch": 97}
{"train_loss": 2.374569322123672, "train_lr": 1.3105467750839792e-06, "train_wd": 0.39979253130132153, "epoch": 98}
{"train_loss": 2.3742457352804416, "train_lr": 1.044445115410006e-06, "train_wd": 0.39997030848577564, "epoch": 99}
